### ISSUE TITLE
Moving NGINX_PORT to Port 81

### ISSUE THREAD

**Issue by bstin** [getumbrel/umbrel#1136]
Hi All. 

I have modified `NGINX_PORT=81` for the `.env` file so that I can run a reverse-proxy with SSL on the host (that is, host is setup to listen on 80/443 and redirects to 127.0.0.1:81).

All is working ok and I can access main Dashboard. 

However one small issue I see is on the "Apps" and "Apps Store" page, my apps are constantly listed as "Starting..."

However, in reality the apps are really running and I can access them ok...but just concerned that this means something else is broken somewhere?

Any pointers?

**Comment by bstin** [comment]
Adding for any searching this issue. 

After more research: The problem has nothing to do with changing ports, it has to do with the Umbrel webapp is trying to connect via http to ports 3004 and 3007 (manager?)

Since the page was loaded as https, most modern web-browsers block "mixed-mode" access (that is prevent you from accessing http content when page was loaded over https). 

I think it is a design failure to do this, and instead these "api calls to other ports" should be handled via a path. That is something like this:

`https://somedomain.tld/manager-api/`, then this "manager-api" path could be rerouted to `:3004` and `:3007` by a reverse-proxy. 

Added to this, I think its is mistake to not expect access to happen via https, the concept that "your local network is trusted" is a bad idea all around (ie. any office environment or environment with wifi will be vulnerable).

**Comment by hkdb** [comment]
I second your opinion on this. What I did which is similar to your approach yet different is that since the nginx container is already a reverse proxy to begin with and docker-compose specifies ./nginx to be mounted as /etc/nginx, I added my own nginx.conf and ssl certs generated by letsencrypt and edited docker-compose.yml to have nginx publish 443 as well. Perhaps adding something to dynamically edit nginx.conf to make everything go through nginx instead of using different ports for direct user access is the way to go?

ie: https://umbrel.domain.com/bluewallet proxies to http://172.21.21.30:3008

I would love to pitch in but judging from some of the relevant closed issues, it seems the maintainers feel pretty strongly about not supporting SSL especially for the beta releases of this solution.
